utils/transcripts.py for chunking preserving previous behaviour as default for #8087

beaterblank opened the pull request:
fixes #8087 
added chunking mechanism for for transcript loaders srt.py and youtubeloader.py so user could optionally load transcript for a certain duration

Commit:
transcripts.py

beaterblank commented:
Hey @baskaryan  @eyurtsev and @rlancemartin why were some checks not sucessfull? any idea of what can I do?
Thank you!

Commit:
reformating

Commit:
Merge branch 'langchain-ai:master' into master

baskaryan commented:
> Hey @baskaryan @eyurtsev and @rlancemartin why were some checks not sucessfull? any idea of what can I do?
> Thank you!
![Screenshot 2023-07-31 at 5 00 37 PM](https://github.com/langchain-ai/langchain/assets/22008038/f674c4f2-7bf7-43ec-9c7c-681e4827f05e)

looks like some linting errors. you can run `make lint` from the libs/langchain directory to see linting issues locally! and `make format` to automatically fix any formatting issues

hwchase17 commented on libs/langchain/langchain/document_loaders/youtube.py:
```diff
@@ -177,7 +179,7 @@ def from_youtube_url(cls, youtube_url: str, **kwargs: Any) -> YoutubeLoader:
         return cls(video_id, **kwargs)
 
-    def load(self) -> List[Document]:
+    def load(self, duration: Optional[float] = None) -> List[Document]:
```
this should not be passed into load

beaterblank replied:
Hey It is an optinal variable and when its not given its defaulted to None and that results in previous behaiviour, that was my intention, I'm not sure how i would be able to achieve that without the duration variable

beaterblank replied:
okay so ill just move the duration to init method and reference it from there got it thank you!

hwchase17 reviewed (changes requested):
this should be an optional parameter and try to maintain backwards compat

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

Commit:
Added backwards compatability 

As per review added the backwards compatability to the load function removing the duration variable and putting it in the init

Commit:
Merge branch 'master' into master

leo-gan commented:
@beaterblank Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

beaterblank commented:
Hey sure yeah, thank you!

Commit:
Merge branch 'master' into master

beaterblank commented:
@leo-gan Hi, the merge issue was regarding updating the branch, and I have updated my branch, 

regarding previous comments I was told that 

 - all configuration information is passed via the initializer by convention 

so I addressed that in my previous commit removing duration from load function and I've put it in the initializer as per the convention

Commit:
Merge branch 'master' into master

leo-gan commented:
@eyurtsev Please, review it. Thanks

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

beaterblank commented:
@eyurtsev please review, Thanks.

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

w4rlock999 commented:
any update on this please? @beaterblank

Commit:
Merge branch 'master' into master

beaterblank commented:
> any update on this please? @beaterblank this is so cool if successfully merged!

@w4rlock999, I'm waiting for it to be reviewed  and merged, I made the requested changes still waiting to be merged.

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

efriis commented on libs/langchain/langchain/document_loaders/srt.py:
```diff
@@ -17,11 +20,30 @@ def __init__(self, file_path: str):
         self.file_path = file_path
 
-    def load(self) -> List[Document]:
+    def load(self, duration: Optional[int] = None) -> List[Document]:
```
same - should be an init param, not a load param

beaterblank replied:
Hello thank you for pointing out.

beaterblank replied:
acb6c8cda2cf594ad592df4c316340218b2ffc33 resolves this

beaterblank replied:
@efriis please review

efriis reviewed (commented):
And please resolve the lint errors :). You should be able to resolve them with `make format` in libs/langchain. You'll need ruff 0.1.5 installed (per the deps)

beaterblank committed:
Merge branch 'master' into master

beaterblank committed:
Merge branch 'master' into master

beaterblank committed:
code refactor and make format

beaterblank committed:
Merge branch 'master' into master

beaterblank committed:
Merge branch 'master' into master
